cli/server.go

```diff
@@ -1047,35 +1071,39 @@ func configureTLS(listener net.Listener, tlsMinVersion, tlsClientAuth, tlsCertFi
 		return nil, xerrors.Errorf("unrecognized tls client auth: %q", tlsClientAuth)
 	}
 
-	if tlsCertFile == "" {
-		return nil, xerrors.New("tls-cert-file is required when tls is enabled")
-	}
-	if tlsKeyFile == "" {
-		return nil, xerrors.New("tls-key-file is required when tls is enabled")
-	}
-
-	certPEMBlock, err := os.ReadFile(tlsCertFile)
+	certs, err := loadCertificates(tlsCertFiles, tlsKeyFiles)
 	if err != nil {
-		return nil, xerrors.Errorf("read file %q: %w", tlsCertFile, err)
+		return nil, xerrors.Errorf("load certificates: %w", err)
 	}
-	keyPEMBlock, err := os.ReadFile(tlsKeyFile)
-	if err != nil {
-		return nil, xerrors.Errorf("read file %q: %w", tlsKeyFile, err)
-	}
-	keyBlock, _ := pem.Decode(keyPEMBlock)
-	if keyBlock == nil {
-		return nil, xerrors.New("decoded pem is blank")
-	}
-	cert, err := tls.X509KeyPair(certPEMBlock, keyPEMBlock)
-	if err != nil {
-		return nil, xerrors.Errorf("create key pair: %w", err)
-	}
-	tlsConfig.GetCertificate = func(chi *tls.ClientHelloInfo) (*tls.Certificate, error) {
-		return &cert, nil
+	tlsConfig.GetCertificate = func(hi *tls.ClientHelloInfo) (*tls.Certificate, error) {
+		// If there's only one certificate, return it.
+		if len(certs) == 1 {
+			return &certs[0], nil
+		}
+
+		// Expensively check which certificate matches the client hello.
+		for _, cert := range certs {
+			cert := cert
+			if err := hi.SupportsCertificate(&cert); err == nil {
+				return &cert, nil
+			}
+		}
+
+		// Return the first certificate if we have one, or return nil so the
+		// server doesn't fail.
+		if len(certs) > 0 {
+			return &certs[0], nil
+		}
+		return nil, nil //nolint:nilnil
 	}
 
+	// Append all certs to the RootCAs list.
 	certPool := x509.NewCertPool()
-	certPool.AppendCertsFromPEM(certPEMBlock)
+	for _, c := range certs {
+		for _, pemBytes := range c.Certificate {
+			certPool.AppendCertsFromPEM(pemBytes)
+		}
+	}
 	tlsConfig.RootCAs = certPool
```

Comment:
I think RootCAs is just for clients. This doesn't make sense for the server does it?

If so, it's just the certs we provide. I think we should drop this.